test/integration/controller/core/clusterqueue_controller_test.go

```diff
@@ -708,28 +709,44 @@ var _ = ginkgo.Describe("ClusterQueue controller", ginkgo.Ordered, ginkgo.Contin
 
 	ginkgo.When("Deleting clusterQueues", func() {
 		var (
-			cq *kueue.ClusterQueue
-			lq *kueue.LocalQueue
+			cq    *kueue.ClusterQueue
+			lq    *kueue.LocalQueue
+			check *kueue.AdmissionCheck
 		)
 
 		ginkgo.BeforeEach(func() {
-			cq = testing.MakeClusterQueue("foo-cq").Obj()
+			check = testing.MakeAdmissionCheck("check").ControllerName("check-controller").Obj()
+			gomega.Expect(k8sClient.Create(ctx, check)).To(gomega.Succeed())
+
+			cq = testing.MakeClusterQueue("foo-cq").AdmissionChecks(check.Name).Obj()
 			lq = testing.MakeLocalQueue("queue", ns.Name).ClusterQueue(cq.Name).Obj()
 			gomega.Expect(k8sClient.Create(ctx, lq)).To(gomega.Succeed())
 			gomega.Expect(k8sClient.Create(ctx, cq)).To(gomega.Succeed())
 		})
 
+		ginkgo.AfterEach(func() {
+			util.ExpectAdmissionCheckToBeDeleted(ctx, k8sClient, check, true)
+		})
+
 		ginkgo.It("Should delete clusterQueues successfully when no admitted workloads are running", func() {
 			util.ExpectClusterQueueToBeDeleted(ctx, k8sClient, cq, true)
 		})
 
 		ginkgo.It("Should be stuck in termination until admitted workloads finished running", func() {
```

Comment:
IMO, we should leave this case without admissionCheck.
So, can you create another case with admissionChecks?

Reply:
In my opinion , the  test is about how the queue is deleted while there are admitted workloads, the way the workloads become admitted is not relevant.

Reply:
It makes sense. `QuotaReserved` is added to workloads even if we don't use two-phase admission.

Reply:
For the purpose of not enlarging the suite indefinitely, I think this is fine.